pkg/render/render_test.go

```diff
@@ -38,6 +38,41 @@ import (
 	rmeta "github.com/tigera/operator/pkg/render/common/meta"
 )
 
+// allCalicoComponents takes the given configuration and returns all the components
+// associated with installing Calico's core, similar to how the core_controller behaves.
+func allCalicoComponents(
```

Comment:
This is a dummy test function that mimics what we used to do for `render.Calico` - I think we can probably just remove this test code after these changes, but wanted to keep it around for testing purposes.

Reply:
Another option would be to move these to be tests in core_controller_test and have the tests be more end-to-end (because it is also through the reconcile). Just a thought.

Reply:
These tests are super horrible and brittle, I am finding out. They have a lot of assertions of the form "expected 30 to be 29" without any indication of which resource was extra / missing.

I think they probably need a heavy re-write, but for now I am trying to keep them in place and get them passing.